File: type/type_multiple_author/src/main/java/com/walhalla/core/domain/db/AppDatabase.java
package com.walhalla.core.domain.db;

import androidx.room.Database;
import androidx.room.RoomDatabase;

import com.walhalla.core.domain.CategoryDao;
import com.walhalla.core.domain.Const;
import com.walhalla.core.domain.db.StatusDao;
import com.walhalla.core.domain.entity.Author;
import com.walhalla.core.domain.entity.Category;
import com.walhalla.core.domain.entity.Status;
import com.walhalla.core.domain.entity.StatusCategoryContract;


@Database(entities = {Author.class, Category.class, Status.class, StatusCategoryContract.class}, version = Const.version, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {

    public abstract CategoryDao categoryDao();

    public abstract StatusDao statusDao();

    public abstract AuthorDao authorDao();
}

File: type/type_single/src/main/java/com/walhalla/core/domain/db/AppDatabase.kt
package com.walhalla.core.domain.db

import androidx.room.Database
import androidx.room.RoomDatabase
import com.walhalla.core.domain.CategoryDao
import com.walhalla.core.domain.Constants
import com.walhalla.core.domain.entity.Category
import com.walhalla.core.domain.entity.Status

@Database(
    entities = [Category::class, Status::class],
    version = Constants.version,
    exportSchema = false
)
abstract class AppDatabase : RoomDatabase() {
    abstract fun categoryDao(): CategoryDao

    abstract fun statusDao(): StatusDao
}
